pytorch_lightning/accelerators/accelerator.py

```diff
@@ -102,11 +103,22 @@ def start_predicting(self, trainer: 'pl.Trainer') -> None:
 
     def pre_dispatch(self, trainer: 'pl.Trainer') -> None:
         """Hook to do something before the training/evaluation/prediction starts."""
+        self.move_optimizer_state()
+
         self.training_type_plugin.pre_dispatch()
         if self.training_type_plugin.setup_optimizers_in_pre_dispatch:
             self.setup_optimizers(trainer)
+
         self.precision_plugin.pre_dispatch()
 
+    def move_optimizer_state(self) -> None:
+        """ Moves the state of the optimizers to the GPU if needed. """
+        for opt in self.optimizers:
+            state = defaultdict(dict)
+            for p, v in opt.state.items():
+                state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, self.root_device)
+            opt.state = state
```

Comment:
n00b q: what's the difference between this and 
```
opt.load_state_dict(move_data_to_device(opt.state_dict(), device=self.root_device))
```

Reply:
The load_dict contains the param_groups.
I am acting only on the state.

Reply:
It was breaking because of this tensor: https://github.com/pytorch/pytorch/blob/master/torch/optim/adagrad.py#L45 left on cpu.